src/widgets/immunizations/vaccination-row.component.tsx

```diff
@@ -0,0 +1,194 @@
+                onClick={() => {
+                  hasExistingDoses(patientImmunization) &&
+                    setToggleOpen(!toggleOpen);
+                }}
+              >
+                <use
+                  xlinkHref={
+                    hasExistingDoses(patientImmunization)
+                      ? toggleOpen
+                        ? "#omrs-icon-chevron-up"
+                        : "#omrs-icon-chevron-down"
+                      : ""
+                  }
+                />
+              </svg>
+            </div>
+            <span>{patientImmunization.vaccineName}</span>
+          </td>
+          <td>
+            <div className={`${styles.alignRight}`}>
+              {getRecentVaccinationText(t, patientImmunization)}
+            </div>
+          </td>
+          <td>
+            <div className={styles.headerAdd}>
+              <button
+                className={`${styles.addButton}`}
+                onClick={() =>
+                  openWorkspaceTab(ImmunizationsForm, "Immunizations Form", [
+                    {
+                      vaccineName: patientImmunization?.vaccineName,
+                      vaccineUuid: patientImmunization?.vaccineUuid,
+                      sequences: patientImmunization?.sequences
+                    }
+                  ])
+                }
+              >
+                +
+              </button>{" "}
+            </div>
+          </td>
+        </tr>
+        {toggleOpen && (
+          <tr
+            id={patientImmunization?.uuid}
+            className={`immunizationSequenceRow ${vaccinationRowStyles.sequenceRow}`}
+          >
+            <td colSpan={4}>
+              <table
+                className={`omrs-type-body-regular immunizationSequenceTable ${vaccinationRowStyles.sequenceTable}`}
+              >
+                <thead>
+                  <tr>
+                    <td>{capitalize(t("sequence", "Sequence"))}</td>
+                    <td>
+                      {capitalize(t("vaccination date", "Vaccination Date"))}
+                    </td>
+                    <td>
+                      {capitalize(t("expiration date", "Expiration Date"))}
+                    </td>
+                    <td></td>
+                  </tr>
+                </thead>
+                <tbody>
+                  {renderSequenceTable(match, t, patientImmunization)}
+                </tbody>
+              </table>
+            </td>
+          </tr>
+        )}
+      </React.Fragment>
+    )
+  );
+}
+
+function getRecentVaccinationText(t, patientImmunization) {
+  if (!hasExistingDoses(patientImmunization)) {
+    return "";
+  }
+  let recentDose = patientImmunization.doses[0];
+  if (hasSequence(patientImmunization)) {
+    return (
+      recentDose.sequenceLabel +
+      " on " +
+      dayjs(recentDose.occurrenceDateTime).format("DD-MMM-YYYY")
+    );
+  }
+  const singleDoseText = t("single dose", "Single Dose");
+  return (
+    singleDoseText +
+    " on " +
```

Comment:
Actually I would not do translation like this. Instead, I would add an argument "date" to the translation string and bring this one in dynamically. That way the exact location (and connection word, e.g., in English "on") would be left to the translation.

Reply:
Just confirming! I should provide `Single Dose on ${recentDose}` as the translated value(for english). And then replace the `{recentDose}` here.

Reply:
You can use variables in your locale string, e.g.,

```
"foo": "this is my translated string for you, {{name}}",
```

which can then be used like

```
t("foo", { name: "Padawan" })
```

Hope that helps!

Reply:
Thanks! I have implemented it. Didn't know there was a default capability to use formats in the react i18n. Just one confirmation, To get it working I had to use `{name}` instead of `{{name}}`. Single curly brackets.